### PR TITLE
Add CLI tests

### PR DESCRIPTION
**Pull request by luc10921**
**Summary or solution description**
Added CLI tests on neo3-boa

**Tests**
https://github.com/CityOfZion/neo3-boa/blob/982065cc0c13e2e25da481e052271fe9f18328ce/boa3_test/tests/cli_tests/test_compile.py#L1-L160

**Platform:**
 - OS: Windows 11 x64
 - Python version: Python 3.9

### CODE REVIEW

**Comment by lllwvlvwlll** [comment]
Task linked: [CU-864e9nnrp Add CLI tests](https://app.clickup.com/t/864e9nnrp)

File: boa3_test/tests/cli_tests/test_compile.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        sc_name = 'smart_contract.py'
+        nef_path, manifest_path = self.get_deploy_file_paths(sc_name)
+        debug_info_path = nef_path.replace('nef', 'nefdbgnfo')
+
+        if os.path.isfile(nef_path):
+            os.remove(nef_path)
+        if os.path.isfile(manifest_path):
+            os.remove(manifest_path)
+        if os.path.isfile(debug_info_path):
+            os.remove(debug_info_path)
```
Comment (meevee98) [inline boa3_test/tests/cli_tests/test_compile.py:51]: Is there a reason to remove these files? Does the test fail if the files exist?

File: boa3_test/test_cli/smart_contract_folder/smart_contract.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from boa3.builtin import env
+from boa3.builtin.compile_time import public
+
+
+@public
+def main() -> str:
+    return env
```
[the remark this answers is not in the document]
Comment (luc10921) [reply boa3_test/test_cli/smart_contract_folder/smart_contract.py:7]: I was using the `smart_contract.py` mostly to guarantee that the folder inside test_cli would exist

File: boa3_test/tests/cli_tests/test_compile.py (hunk 1 above)
Comment (luc10921) [reply boa3_test/tests/cli_tests/test_compile.py:51]: It was to guarantee that the CLI command would generate the files if something went wrong with the compilation

**Pull request merged**